### DISCUSSION TITLE
Future Updates to http-cors

### DISCUSSION THREAD

**Discussion by cspray** [labrador-kennel/http-cors#36] [Announcements]
It has been nearly 2 years since this library had a release made. With the release of fibers in PHP a lot of improvements and changes have been made to the underlying [Amphp]() packages this library relies on. Additionally, this library never hit a stable 1.0 release as it should have. The rest of this document details the path I plan on taking in upgrading Amp support and getting to a 1.0 release.

For clarity, the current release is **0.3.1**

## 0.4.0

- Add support for amphp/http-server v3, drops support for previous versions of http-server
- Drop support for PHP 7

## 0.5.0

- Deprecates the `Cspray\Labrador\Http\Cors` namespace in favor of the `Labrador\Http\Cors` namespace.
- Adds aliases for desired class names to allow starting transition.

## 0.6.0

- Removes the `Cspray\Labrador` top-level namespace in favor of `Labrador` top-level namespace.
- Moves the package from `cspray/labrador-http-cors` to `labrador-kennel/http-cors`

## 0.7.0

- Add ability to pass a `Psr\Log\LoggerInterface` to the middleware to log pertinent information.

## 1.0

- Update documentation to reflect most recent code and changes. Add more docs where might be missing.

**Comment by Nek-** [comment]
Lovely, thank you for this post. It's unusual in libraries but super useful to understand your vision.
